src/reducers/ui-state-updaters.js

```diff
@@ -201,6 +206,20 @@ export const INITIAL_UI_STATE = {
 };
 
 /* Updaters */
+
+export const initUIStateUpdater = (state, action) => {
+  const isTokenValid = validateToken((action.payload || {}).mapboxApiAccessToken);
```

Comment:
do a sync validation of the token and update notification and modal.
If the token is not valid we also hide the data molda